### PR TITLE
Improve generation of exit and error code files

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

Auto-regenerate exit code and error code files in non-maintainer mode, too.
Previously these files were only generated in maintainer mode.
This simplifies and unifies the build for maintainer mode in non-maintainer mode w.r.t. these files.
Also added clang-format instructions so that clang-format does not reformat these files or complains about them.
This does not need to be backported.

- [ ] :hankey: Bugfix
- [ ] :pizza: New feature
- [ ] :fire: Performance improvement
- [x] :hammer: Refactoring/simplification

### Checklist

- [ ] Tests
  - [ ] **Regression tests**
  - [ ] C++ **Unit tests**
  - [ ] **integration tests**
  - [ ] **resilience tests**
- [ ] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [ ] Backports
  - [ ] Backport for 3.9: -
  - [ ] Backport for 3.8: -
  - [ ] Backport for 3.7: -

#### Related Information

- [ ] Docs PR: 
- [ ] Enterprise PR:
- [ ] GitHub issue / Jira ticket:
- [ ] Design document:

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1314,65 +1314,63 @@ if (NOT(SKIP_PACKAGING))
...
-    lib/Basics/voc-errors.h
-    lib/Basics/error-registry.h
-    js/common/bootstrap/errors.js
+set(ERROR_FILES_GEN)
+set(ERRORS_DAT lib/Basics/errors.dat)
+
+foreach (m IN LISTS ERROR_FILES)
+  add_custom_command(
+    OUTPUT ${CMAKE_SOURCE_DIR}/${m}
+    COMMAND ${PYTHON_EXECUTABLE} ./utils/generateErrorfile.py ./${ERRORS_DAT} ./${m}.tmp
```
Comment (dothebart) [inline CMakeLists.txt:1335]: prepend `${CMAKE_SOURCE_DIR}/` pls instead of using `./` 
else LGTM.

Comment (jsteemann) [reply CMakeLists.txt:1335]: Note: This is simply the old code without the `if (USE_MAINTAINER_MODE)` around it.

Comment (jsteemann) [reply CMakeLists.txt:1335]: Btw the scope of the command is 
```
WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
```

```
  add_custom_command(
    OUTPUT ${CMAKE_SOURCE_DIR}/${m}
    COMMAND ${PYTHON_EXECUTABLE} ./utils/generateErrorfile.py ./${ERRORS_DAT} ./${m}.tmp
    COMMAND ${CMAKE_COMMAND} -E copy_if_different ./${m}.tmp ./${m}
    COMMAND ${CMAKE_COMMAND} -E remove ./${m}.tmp
    DEPENDS ${CMAKE_SOURCE_DIR}/${ERRORS_DAT} ./utils/generateErrorfile.py
    WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
    COMMENT  "Building errors files ${m}"
    VERBATIM
  )
```
So using `./` seems to be fine.

File: utils/generateErrorfile.py [hunk 2]
```diff
@@ -67,26 +67,26 @@ def errorName(error):
...
 """ \
            + wiki
 
   # print individual errors
   for e in errors:
     header = header\
+           + "\n"\
            + "/// " + e[1] + ": " + e[0] + "\n"\
            + wrap(e[2], 80, 0, 0, "/// \"") + "\"\n"\
            + wrap(e[3], 80, 0, 0, "/// ") + "\n"\
```
Comment (Simran-B) [inline utils/generateErrorfile.py:88]: Why does clang-format not complain about the `///` here?

File: utils/generateExitCodesFiles.py [hunk 3]
```diff
@@ -109,27 +109,24 @@ def genNSISFile(errors, filename):
...
+           + "/* clang-format off */\n"\
+           + wiki
 
   # print individual errors
   for e in errors:
     header = header\
+           + "\n"\
            + "/// " + e[1] + ": " + e[0] + "\n"\
            + wrap(e[2], 80, 0, 0, "/// ") + "\n"\
            + wrap(e[3], 80, 0, 0, "/// ") + "\n"\
```
Comment (Simran-B) [inline utils/generateExitCodesFiles.py:128]: Why does clang-format not complain about `///` here?

Comment (jsteemann) [reply utils/generateExitCodesFiles.py:128]: @Simran-B : please leave the PR alone. I already added a fix for clang-format complaining about it.

**Review by maierlars — APPROVED** [review]

**Pull request merged**